src/backend/user/user.service.ts

```diff
@@ -83,23 +82,27 @@ export async function syncUserWithGithub(credentials: {
   return null
 }
 
-export async function createUser(payload: CreateUserPayload) {
-  const createdUser = await prisma.user.create({
-    data: {
-      email: payload.email,
-      avatarUrl: payload.image,
-      githubDetails: {
-        create: {
-          username: payload.githubUsername,
-        },
-      },
-    },
-    include: {
-      githubDetails: true,
+export async function createGithubDetails(
+  userId: string,
+  githubUsername: string,
+) {
+  const existingDetails = await prisma.gitHubDetails.findUnique({
+    where: {
+      userId: userId,
     },
   })
 
-  return createdUser
+  if (existingDetails) {
+    // GitHub details already exist for this user, so we don't need to create again
+    return
```

Comment:
**nit:** Let's return found `existingDetails` instead of undefined

Reply:
This method has been removed because the functionality is now handled by the custom createUser method.
Refer to the implementation in the following commit: https://github.com/nerdbord/good-dev-hunting-app/pull/263/commits/110e303c9f00429dda5cf014708327f86c3711be